src/main/java/org/skriptlang/skript/bukkit/recipes/MutableRecipe.java

```diff
@@ -0,0 +1,411 @@
+package org.skriptlang.skript.bukkit.recipes;
+
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.*;
+import org.bukkit.inventory.recipe.CookingBookCategory;
+import org.bukkit.inventory.recipe.CraftingBookCategory;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.recipes.RecipeUtils.RecipeType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Mutable Recipe class allowing data to be set before creation of a recipe.
```

Comment:
you should add that this is a this is used as a wrapper for all the different types of recipes